JavaScript/JavaScriptSDK.Tests/CheckinTests/SendBuffer.tests.ts

```diff
@@ -0,0 +1,420 @@
+    private getSessionStorageSendBuffer: (emitLineDelimitedJson?: boolean) => Microsoft.ApplicationInsights.SessionStorageSendBuffer;
+
+    public testCleanup() {
+        Microsoft.ApplicationInsights.DataLossAnalyzer.enabled = false;
+    }
+
+    public testInitialize() {
+        if (Microsoft.ApplicationInsights.Util.canUseSessionStorage()) {
+            sessionStorage.clear();
+        }
+
+        var config: Microsoft.ApplicationInsights.ISenderConfig = {
+            emitLineDelimitedJson: () => false,
+            storeSendBufferInSessionStorage: () => false,
+            endpointUrl: () => null,
+            maxBatchSizeInBytes: () => null,
+            maxBatchInterval: () => null,
+            disableTelemetry: () => null
+        };
+
+        this.getArraySendBuffer = (emitLineDelimitedJson?: boolean) => {
+            if (emitLineDelimitedJson) {
+                config.emitLineDelimitedJson = () => emitLineDelimitedJson;
+            }
+
+            return new Microsoft.ApplicationInsights.ArraySendBuffer(config);
+        }
+
+        this.getSessionStorageSendBuffer = (emitLineDelimitedJson?: boolean) => {
+            if (emitLineDelimitedJson) {
+                config.emitLineDelimitedJson = () => emitLineDelimitedJson;
+            }
+
+            return new Microsoft.ApplicationInsights.SessionStorageSendBuffer(config);
+        }
+    }
+
+    private BUFFER_KEY = "AI_buffer";
+    private SENT_BUFFER_KEY = "AI_sentBuffer";
+
+    public registerTests() {
+        // ArraySendBuffer tests
+
+        this.testCase({
+            name: "ArraySendBuffer: initialize",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // verify
+                Assert.equal(0, buffer.count(), "new buffer should be empty");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: can enqueue and clear the buffer",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                buffer.enqueue("");
+
+                // verify
+                Assert.equal(1, buffer.count(), "one item expected");
+
+                // act
+                buffer.enqueue("");
+
+                // verify
+                Assert.equal(2, buffer.count(), "two items expected");
+
+                //act
+                buffer.clear();
+
+                // verify
+                Assert.equal(0, buffer.count(), "buffer should be empty");
+
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: can clear empty buffer",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                buffer.clear();
+
+                // verify
+                Assert.equal(0, buffer.count(), "buffer should be empty");
```

Comment:
It might be good to also assert that it was empty to start with since that is a starting assumption.